packages/common/utils/src/groupBy.ts

```diff
@@ -1,5 +1,5 @@
 /** @tossdocs-ignore */
-export function groupBy<T>(data: T[], createKey: (item: T) => string) {
+export function groupBy<T>(data: T[], createKey: (item: T) => string | number) {
   return data.reduce((result: Record<string, T[]>, current) => {
```

Comment:
> Reduce function's initial value is object, object only allow key of string type.
So we don't need to add number type to result: Record<string, T[]>

You can define an object with number keys and it can be accessed by both string and number.

```typescript
const obj = { 1: 'foo' };

console.log(obj[1]); // 'foo'
console.log(obj['1']); // 'foo'
```

Reply:
You are right, but I think the below code is not very hard to write, and we would like to keep the implementation as simple as possible, to keep the code maintainable.

```tsx
groupBy(items, item => item.id.toString())
```

Also, numbers are not the only type we should consider. We might have to support boolean as well, if we want to group them by true or false. These decisions might make the code complex, so we want to keep the code as simple as possible, and provide the very basic functionality. (Users of code might wrap our implementation)

Reply:
@HoseungJang @raon0211 
I understand your opinion well. Thanks for the detail comments!